e2e-tests/insurance/cypress/e2e/journeys/your-buyer/connection-to-the-buyer/connection-to-the-buyer-page.spec.js

```diff
@@ -0,0 +1,136 @@
+
+const {
+  WORKING_WITH_BUYER: {
+    CONNECTED_WITH_BUYER,
+    CONNECTION_WITH_BUYER_DESCRIPTION,
+  },
+} = FIELD_IDS;
+
+const {
+  YOUR_BUYER: { COMPANY_OR_ORGANISATION, CONNECTION_TO_THE_BUYER, WORKING_WITH_BUYER },
+} = ROUTES.INSURANCE;
+
+const FIELDS = YOUR_BUYER_FIELDS.WORKING_WITH_BUYER;
+
+const baseUrl = Cypress.config('baseUrl');
+
+context('Insurance - Your Buyer - Connection with the buyer - As an exporter, I want to provide the details on trading history with the buyer of my export trade, So that UKEF can gain clarity on whether I have  trading history with the buyer as part of due diligence', () => {
+  let referenceNumber;
+  let url;
+  let workingWithBuyerUrl;
+
+  before(() => {
+    cy.completeSignInAndGoToApplication({}).then(({ referenceNumber: refNumber }) => {
+      referenceNumber = refNumber;
+
+      cy.startInsuranceYourBuyerSection();
+
+      cy.completeAndSubmitCompanyOrOrganisationForm({});
+
+      url = `${baseUrl}${INSURANCE_ROOT}/${referenceNumber}${CONNECTION_TO_THE_BUYER}`;
+      workingWithBuyerUrl = `${baseUrl}${INSURANCE_ROOT}/${referenceNumber}${WORKING_WITH_BUYER}`;
+
+      cy.assertUrl(url);
+    });
+  });
+
+  beforeEach(() => {
+    cy.saveSession();
+  });
+
+  after(() => {
+    cy.deleteApplication(referenceNumber);
+  });
+
+  it('renders core page elements', () => {
+    cy.corePageChecks({
+      pageTitle: CONTENT_STRINGS.PAGE_TITLE,
+      currentHref: `${INSURANCE_ROOT}/${referenceNumber}${CONNECTION_TO_THE_BUYER}`,
+      backLink: `${INSURANCE_ROOT}/${referenceNumber}${COMPANY_OR_ORGANISATION}`,
+    });
+  });
+
+  describe('page tests', () => {
+    beforeEach(() => {
+      cy.navigateToUrl(url);
+    });
+
+    it('renders a heading caption', () => {
+      cy.checkText(headingCaption(), CONTENT_STRINGS.HEADING_CAPTION);
+    });
+
+    describe(CONNECTED_WITH_BUYER, () => {
+      it('renders a hint', () => {
+        cy.checkText(yesNoRadioHint(), CONTENT_STRINGS.HINT);
+      });
+
+      it('renders `yes` radio button', () => {
+        yesRadio().input().should('exist');
+
+        cy.checkText(yesRadio().label(), FIELD_VALUES.YES);
+
+        cy.checkRadioInputYesAriaLabel(CONTENT_STRINGS.PAGE_TITLE);
+      });
+
+      it('renders `no` radio button', () => {
+        noRadio().input().should('exist');
+
+        cy.checkText(noRadio().label(), FIELD_VALUES.NO);
+
+        cy.checkRadioInputNoAriaLabel(CONTENT_STRINGS.PAGE_TITLE);
+      });
+    });
+
+    describe(CONNECTION_WITH_BUYER_DESCRIPTION, () => {
+      const fieldId = CONNECTION_WITH_BUYER_DESCRIPTION;
+
+      describe(`${CONNECTED_WITH_BUYER} not selected`, () => {
+        it('should not render a label', () => {
+          field(fieldId).label().should('not.be.visible');
+        });
+
+        it('should not render a visible input', () => {
+          field(fieldId).textarea().should('not.be.visible');
+        });
+      });
+
+      describe(`${CONNECTED_WITH_BUYER} yes radio selected`, () => {
```

Comment:
Should we/could we align with `about-goods-or-services.spec.js` ? e.g it's more like a story

```js
describe(`when clicking the 'yes' ${FINAL_DESTINATION_KNOWN} radio`, () => { ...
```